src/libraries/Microsoft.PowerFx.Core/Texl/Builtins/Collect.cs

```diff
@@ -0,0 +1,467 @@
+
+        public override bool RequiresDataSourceScope => true;
+
+        public virtual bool IsScalar => false;
+
+        public override bool CanSuggestInputColumns => true;
+
+        public override bool MutatesArg0 => true;
+
+        public override RequiredDataSourcePermissions FunctionPermission => RequiredDataSourcePermissions.Create;
+
+        /// <summary>
+        /// Since Arg1 and Arg2 depends on type of Arg1 return false for them.
+        /// </summary>
+        public override bool TryGetTypeForArgSuggestionAt(int argIndex, out DType type)
+        {
+            if (argIndex == 1 || argIndex == 2)
+            {
+                type = default;
+                return false;
+            }
+
+            return base.TryGetTypeForArgSuggestionAt(argIndex, out type);
+        }
+
+        public override bool ArgMatchesDatasourceType(int argNum)
+        {
+            return argNum >= 1;
+        }
+
+        public override bool IsLazyEvalParam(int index, Features features)
+        {
+            // First argument to mutation functions is Lazy for datasources that are copy-on-write.
+            // If there are any side effects in the arguments, we want those to have taken place before we make the copy.
+            return index == 0;
+        }
+
+        public CollectFunction()
+            : this("Collect", TexlStrings.AboutCollect)
+        {
+        }
+
+        protected CollectFunction(string name, TexlStrings.StringGetter description)
+            : base(name, description, FunctionCategories.Behavior, DType.EmptyRecord, 0, 2, int.MaxValue, DType.EmptyTable)
+        {
+        }
+
+        public override IEnumerable<TexlStrings.StringGetter[]> GetSignatures()
+        {
+            // !!! TODO
+            //yield return new[] { CanvasStringResources.CollectArg1, CanvasStringResources.CollectArg2 };
+            //yield return new[] { CanvasStringResources.CollectArg1, CanvasStringResources.CollectArg2, CanvasStringResources.CollectArg2 };
+            //yield return new[] { CanvasStringResources.CollectArg1, CanvasStringResources.CollectArg2, CanvasStringResources.CollectArg2, CanvasStringResources.CollectArg2 };
+
+            yield return new[] { TexlStrings.CollectDataSourceArg, TexlStrings.CollectItemArg };
+        }
+
+        public override IEnumerable<TexlStrings.StringGetter[]> GetSignatures(int arity)
+        {
+            if (arity > 2)
+            {
+                return GetGenericSignatures(arity, TexlStrings.CollectDataSourceArg, TexlStrings.CollectItemArg);
+            }
+
+            return base.GetSignatures(arity);
+        }
+
+        public virtual DType GetCollectedType(Features features, DType argType)
+        {
+            Contracts.Assert(argType.IsValid);
+
+            return argType;
+        }
+
+        public bool TryGetUnifiedCollectedTypeCanvas(TexlNode[] args, DType[] argTypes, IErrorContainer errors, Features features, out DType collectedType)

```

Comment:
>public [](http://example.com/codeflow?start=8&length=6)

This doesn't seem to be used in any other class; if that's the case, please make it private. Same for V1